src/annotator/sidebar.coffee

```diff
@@ -16,38 +19,51 @@ module.exports = class Sidebar extends Host
     TextSelection: {}
     BucketBar:
       container: '.annotator-frame'
-    Toolbar:
-      container: '.annotator-frame'
 
   renderFrame: null
   gestureState: null
 
   constructor: (element, config) ->
     if config.theme == 'clean' || config.externalContainerSelector
       delete config.pluginClasses.BucketBar
+
+    # TODO - Make this work again.
     if config.externalContainerSelector
       delete config.pluginClasses.Toolbar
 
     super
 
-    this.hide()
-
     if config.openSidebar || config.annotations || config.query || config.group
       this.on 'panelReady', => this.show()
 
     if @plugins.BucketBar?
       @plugins.BucketBar.element.on 'click', (event) => this.show()
 
-    if @plugins.Toolbar?
-      @toolbarWidth = @plugins.Toolbar.getWidth()
-      if config.theme == 'clean'
-        @plugins.Toolbar.disableMinimizeBtn()
-        @plugins.Toolbar.disableHighlightsBtn()
-        @plugins.Toolbar.disableNewNoteBtn()
-      else
-        @plugins.Toolbar.disableCloseBtn()
+    toolbarContainer = document.createElement('div')
+    @toolbar = new ToolbarController(toolbarContainer, {
+      createAnnotation: => this.createAnnotation()
+      setSidebarOpen: (open) =>
+        if open
+          this.show()
+        else
+          this.hide()
+      setHighlightsVisible: (show) => this.setAllVisibleHighlights(show)
+    })
+    @toolbar.useMinimalControls = config.theme == 'clean'
```

Comment:
Just wondering if config could be accessed from within the toolbar itself instead of being passed in?

Reply:
I'm not quite sure I've understood correctly here, but the way I've set this PR up, the toolbar widget doesn't receive the global configuration options any more. Instead the sidebar is responsible for mapping global config options to toolbar config.